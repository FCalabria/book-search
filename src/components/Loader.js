import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHeartBroken } from '@fortawesome/free-solid-svg-icons'
import {LanguageConsumer} from '../language-context'

function Loader(props) {
  return (
    <LanguageConsumer>
    {({language}) => (
      <div className="my-auto mx-3 text-center flex flex-col items-center">
        <div className="text-lg font-light">{props.error ? language.searchError : language.searching} {props.searchTerm}</div>
        {
          props.error
            ? <div className="text-pink-500 mt-3"><FontAwesomeIcon icon={faHeartBroken} size="4x"/></div>
            : <div className="lds-heart"><div></div></div>
        }
    </div>
    )}
    </LanguageConsumer>
  )
}

export default Loader