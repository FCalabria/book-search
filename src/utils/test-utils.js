import React from 'react'
import { render } from '@testing-library/react'
import {LanguageProvider, languages} from '../language-context'


const AllTheProviders = ({ children }) => {
  return (
    <LanguageProvider value={{
      language: languages.ES,
      keys: Object.keys(languages),
      selected: 'ES',
      toggleLanguage: () => {}
    }}>
      {children}
    </LanguageProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
