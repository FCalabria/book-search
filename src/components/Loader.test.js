import React from 'react';
import { render } from '../../tests/test-utils';
import Loader from './Loader';

describe('Loader', () => {
  const defaultProps = {
    searchTerm: 'Some book'
  }
  let baseContainer
  beforeEach(() => {
    baseContainer = render(<Loader {...defaultProps} />).container;
  })
  test('should render correctly when there is no error', () => {
    expect(baseContainer.firstChild).toMatchSnapshot();
  });
  test('should render correctly when error exists', () => {
    const { container } = render(<Loader {...defaultProps} error={true}/>);
    expect(baseContainer.firstChild).toMatchDiffSnapshot(container.firstChild);
  })
})
