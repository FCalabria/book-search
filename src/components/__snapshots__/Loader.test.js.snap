// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly when error exists 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -2,15 +2,29 @@
    class=\\"my-auto mx-3 text-center flex flex-col items-center\\"
  >
    <div
      class=\\"text-lg font-light\\"
    >
-     Buscando
+     No hemos encontrado nada con
       
      Some book
    </div>
    <div
-     class=\\"lds-heart\\"
+     class=\\"text-pink-500 mt-3\\"
    >
-     <div />
+     <svg
+       aria-hidden=\\"true\\"
+       class=\\"svg-inline--fa fa-heart-broken fa-w-16 fa-4x \\"
+       data-icon=\\"heart-broken\\"
+       data-prefix=\\"fas\\"
+       focusable=\\"false\\"
+       role=\\"img\\"
+       viewBox=\\"0 0 512 512\\"
+       xmlns=\\"http://www.w3.org/2000/svg\\"
+     >
+       <path
+         d=\\"M473.7 73.8l-2.4-2.5c-46-47-118-51.7-169.6-14.8L336 159.9l-96 64 48 128-144-144 96-64-28.6-86.5C159.7 19.6 87 24 40.7 71.4l-2.4 2.4C-10.4 123.6-12.5 202.9 31 256l212.1 218.6c7.1 7.3 18.6 7.3 25.7 0L481 255.9c43.5-53 41.4-132.3-7.3-182.1z\\"
+         fill=\\"currentColor\\"
+       />
+     </svg>
    </div>
  </div>"
`;

exports[`Loader should render correctly when there is no error 1`] = `
<div
  class="my-auto mx-3 text-center flex flex-col items-center"
>
  <div
    class="text-lg font-light"
  >
    Buscando
     
    Some book
  </div>
  <div
    class="lds-heart"
  >
    <div />
  </div>
</div>
`;
